pyproject.toml

```diff
@@ -73,3 +73,29 @@ filterwarnings = [
     "ignore::FutureWarning:stravalib.*",
     "ignore::DeprecationWarning:stravalib.*"
 ]
+
+[tool.mypy]
+python_version = "3.8"
+follow_imports = "silent"
+warn_redundant_casts = true
+warn_unused_ignores = true
+disallow_any_generics = true
+check_untyped_defs = true
+no_implicit_reexport = true
+warn_unreachable = true
+disallow_untyped_defs = true
+
+plugins = [
+  "pydantic.mypy"
+]
+
+files = [
```

Comment:
this is great that we can limit what mypy looks at. what i wasn't sure about is whether our pre-commit.ci bot will follow this configuration (it might). it seems to want to run on all files which is what i was worried about with CI. We could test that on this PR if [you wish to add the config to the pre-commit.config.yaml file](https://github.com/pre-commit/mirrors-mypy).